packages/mui-base/src/NoSsr/NoSsr.tsx

```diff
@@ -1,17 +1,27 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import { exactProp, unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/utils';
+import { NoSsrProps } from './NoSsr.types';
 
 /**
  * NoSsr purposely removes components from the subject of Server Side Rendering (SSR).
  *
  * This component can be useful in a variety of situations:
- * - Escape hatch for broken dependencies not supporting SSR.
- * - Improve the time-to-first paint on the client by only rendering above the fold.
- * - Reduce the rendering time on the server.
- * - Under too heavy server load, you can turn on service degradation.
+ *
+ * *   Escape hatch for broken dependencies not supporting SSR.
+ * *   Improve the time-to-first paint on the client by only rendering above the fold.
+ * *   Reduce the rendering time on the server.
+ * *   Under too heavy server load, you can turn on service degradation.
+ *
+ * Demos:
+ *
+ * - [No SSR](https://mui.com/base/react-no-ssr/)
+ *
+ * API:
+ *
+ * - [NoSsr API](https://mui.com/base/api/no-ssr/)
  */
-function NoSsr(props) {
+const NoSsr = (props: NoSsrProps) => {
```

Comment:
We are going to introduce an ESLint rule that forbids defining components using arrow functions (see #34644). Please change it to an ordinary function.

```suggestion
function NoSsr(props: NoSsrProps) {
```